### PR TITLE
Removed lwip

### PR DESCRIPTION
**Pull request by CarlWachter**
Removed lwip and replaced all references with interfaces to hermit networking.

Should be merged with hermit-os/hermit-playground#27 to avoid breaking hermit-os/hermit-gcc, as it only pulls the latest commit from newlib.

### CODE REVIEW

File: newlib/libc/sys/hermit/include/sys/socket.h [hunk 1]
```diff
@@ -1,33 +1,186 @@
...
+#define sys_setsockopt        setsockopt
+//#define sys_getsockopt        getsockopt
+#define sys_getpeername       getpeername
+#define sys_freeaddrinfo      freeaddrinfo
+#define sys_getaddrinfo       getaddrinfo
+#define sys_send              send
+#define sys_shutdown_socket   shutdown
+#define sys_recv              recv
+#define sys_sendto            sendto
+#define sys_recvfrom          recvfrom */
```
Comment (mkroening) [inline newlib/libc/sys/hermit/include/sys/socket.h:59]:
```suggestion
```

File: newlib/libc/sys/hermit/include/sys/socket.h [hunk 2]
```diff
@@ -1,33 +1,186 @@
...
+//#define getsockopt(s,level,optname,opval,optlen)     sys_getsockopt(s,level,optname,opval,optlen)
+/* #define getpeername(s,name,namelen)                   sys_getpeername(s,name,namelen) */
+//#define freeaddrinfo(ai)                             sys_freeaddrinfo(ai)
+/* #define send(s,mem,len,_flags)                        sys_send(s,mem,len,_flags)
+#define shutdown(s,how)                               sys_shutdown_socket(s,how)
+#define recv(fd,buf,len,flags)                        sys_recv(fd,buf,len,flags)
+#define sendto(socket,buf,len,flags,addr,addrlen)     sys_sendto(s,dataptr,size,flags,to,tolen)
+#define recvfrom(socket,buf,len,flags,addr,addrlen)   sys_recvfrom(s,mem,len,flags,from,fromlen) */
+
+//#define closesocket(s)                            sys_close(s)
```
Comment (mkroening) [inline newlib/libc/sys/hermit/include/sys/socket.h:116]:
```suggestion
```

Is there some specific reason for these comments?

File: newlib/libc/sys/hermit/Makefile.am [hunk 3]
```diff
@@ -4,7 +4,7 @@ AM_CCASFLAGS = $(INCLUDES)
  
 noinst_LIBRARIES = lib.a
 
-lib_a_SOURCES = chown.c clock_getres.c clock_gettime.c clock_nanosleep.c clock_settime.c close.c environ.c errno.c execve.c _exit.c fork.c fstat.c getpagesize.c getpid.c gettod.c isatty.c kill.c link.c lseek.c nanosleep.c open.c read.c readlink.c sbrk.c sched.c setitimer.c signal.c stat.c symlink.c sysconf.c times.c unlink.c wait.c write.c context.S makecontext.c
+lib_a_SOURCES = chown.c clock_getres.c clock_gettime.c clock_nanosleep.c clock_settime.c close.c environ.c errno.c execve.c _exit.c fork.c fstat.c getpagesize.c getpid.c gettod.c isatty.c kill.c link.c lseek.c nanosleep.c open.c read.c readlink.c sbrk.c sched.c setitimer.c signal.c stat.c symlink.c sysconf.c times.c unlink.c wait.c write.c context.S makecontext.c inet_addr.c inet_ntoa.c inet_ntop.c
```
Comment (mkroening) [inline newlib/libc/sys/hermit/Makefile.am:7]: `inet_pton.c` seems missing.

**Review by mkroening — CHANGES_REQUESTED** [review]
Thanks! Where do `inet_*.c` and friends come from?

**Comment by CarlWachter** [comment]
> Thanks! Where do `inet_*.c` and friends come from?

It's from the debian source code. Ex.:
https://sources.debian.org/data/main/g/glibc/2.28-10/inet/inet_ntoa.c

File: newlib/libc/sys/hermit/Makefile.in
Comment (mkroening) [inline newlib/libc/sys/hermit/Makefile.in:1]: This file is out of date and needs to be regenerated with automake.

**Review by mkroening — CHANGES_REQUESTED** [review]
> > Thanks! Where do `inet_*.c` and friends come from?
> 
> It's from the debian source code. Ex.: https://sources.debian.org/data/main/g/glibc/2.28-10/inet/inet_ntoa.c

Do we actually need any new files besides `inet_ntoa.c`? `Makefile.in` does not have other new files.

Instead of taking them from glibc, can we just use [`newlib/libc/sys/linux/net/inet_ntoa.c`](https://github.com/hermit-os/newlib/blob/7db17585c746d59b539df6fef738b22573d0636b/newlib/libc/sys/linux/net/inet_ntoa.c)?

If we really need to add new files from glibc, please add a [SPDX License ID](https://spdx.dev/learn/handling-license-info/) and add a link to the source. For example:

```c
/* SPDX-License-Identifier: LGPL-2.1-or-later */
/* From https://sourceware.org/git/?p=glibc.git;a=blob;f=inet/inet_ntoa.c;h=341d8591c30b88d4415898d93ae0dcbd7bf4e3d6;hb=46b5e98ef6f1b9f4b53851f152ecb8209064b26c */
```

**Comment by CarlWachter** [comment]
> Do we actually need any new files besides `inet_ntoa.c`? `Makefile.in` does not have other new files.

True, i removed the others. If they are needed in the future, they are easy to add back in.

> Instead of taking them from glibc, can we just use [`newlib/libc/sys/linux/net/inet_ntoa.c`](https://github.com/hermit-os/newlib/blob/7db17585c746d59b539df6fef738b22573d0636b/newlib/libc/sys/linux/net/inet_ntoa.c)?

I tried that and it led to a lot of issues, since one linux file typically seeks to have 20 other linux files included with it.

> If we really need to add new files from glibc, please add a [SPDX License ID](https://spdx.dev/learn/handling-license-info/) and add a link to the source. For example:
> 
> ```c
> /* SPDX-License-Identifier: LGPL-2.1-or-later */
> /* From https://sourceware.org/git/?p=glibc.git;a=blob;f=inet/inet_ntoa.c;h=341d8591c30b88d4415898d93ae0dcbd7bf4e3d6;hb=46b5e98ef6f1b9f4b53851f152ecb8209064b26c */
> ```
Done

File: newlib/libc/sys/hermit/Makefile.in
Comment (CarlWachter) [reply newlib/libc/sys/hermit/Makefile.in:1]: Resolved, by changing Makefile.am back to same version as Makefile.in

**Review by mkroening — APPROVED** [review]

**Pull request merged**